docs/MigrationGuide.md

```diff
@@ -0,0 +1,118 @@
+}
+```
+
+### 2. Update the initialization code
+We won't cover all the details here as most of the configuration can be set using the builders fluent methods. However, the main difference is that the new SDK requires an `Application` context to be passed to the `Unleash` constructor. This is necessary to monitor the network connectivity and respect the Android lifecycle (usually it can be injected if using hilt with `@ApplicationContext context`).
+ 
+#### Unleash context initialization
+The main differences are:
+1. The application name is no longer configurable through the context as it is unique across the application. The `appName` should be set using the `UnleashConfig` builder.
+2. The instance ID is no longer configurable. The SDK will generate a unique instance ID for each instance.
+3. Update the import statements to use the new SDK classes.
+
+```kotlin
+val unleashContext = UnleashContext.newBuilder()
+    // .appName("unleash-android") // remove this line
+    // .instanceId("main-activity-unleash-demo-${Random.nextLong()}") // remove this line
+    .userId("unleash_demo_user")
+    .sessionId(Random.nextLong().toString())
+    .build()
+```
+
+#### Unleash configuration
+The main differences are:
+1. Metrics are enabled by default.
+2. App name is now a mandatory parameter to the builder.
+3. Instance id is no longer configurable.
+4. The polling mode is now a polling strategy with a fluent API.
+5. The metrics interval is now part of the metrics strategy with a fluent API.
+
+**Old SDK**
+```kotlin
+UnleashConfig.newBuilder()
+    .appName("unleash-android")
+    .instanceId("unleash-android-${Random.nextLong()}")
+    .enableMetrics()
+    .clientKey("xyz")
+    .proxyUrl("https://eu.app.unleash-hosted.com/demo/api/frontend")
+    .pollingMode(
+        PollingModes.autoPoll(
+            autoPollIntervalSeconds = 15
+        ) {
+
+        }
+    )
+    .metricsInterval(5000)
+    .build()
+```
+
+**New SDK**
+```kotlin
+UnleashConfig.newBuilder("unleash-android")
+    .clientKey("xyz")
+    .proxyUrl("https://eu.app.unleash-hosted.com/demo/api/frontend")
+    .pollingStrategy.interval(15000)
+    .metricsStrategy.interval(5000)
+    .build()
+```
+
+#### Creating the Unleash instance
+The previous SDK used a builder to construct the Unleash instance while the new SDK relies on constructor parameters. There are also other meaningful changes:
+
+1. The new SDK does not start automatically. You need to call `unleash.start()` to start the polling and metrics collection.
+2. The new SDK accepts event listeners at the constructor level or as parameters when calling `unleash.start()` (you can also edit your config object setting `delayedInitialization` to false).
+3. The interface `UnleashClientSpec` is now `Unleash`.
+
+```kotlin
+UnleashClient.newBuilder()
+    .unleashConfig(unleashConfig)
+    .cache(InMemoryToggleCache()) // This was optional in the previous SDK and in the new one so we are ignoring it
```

Comment:
I don't understand what the comment means. Why are you ignoring it?

Reply:
I was just trying to stay close to the code of the sample app that was including this cache implementation. But there's no point in doing that as the default implementation is the same (if you don't set cache). So I thought just commenting that in the code but I also didn't quite like it. I'll just remove this line